components/calendar/src/provider.rs

```diff
@@ -33,11 +34,11 @@ pub struct EraStartDate {
     feature = "provider_serde",
     derive(serde::Serialize, serde::Deserialize)
 )]
-#[zerofrom(cloning_zf)]
-// TODO (#1393) Make this zero-copy
-pub struct JapaneseErasV1 {
-    pub dates_to_historical_eras: LiteMap<EraStartDate, TinyStr16>,
-    pub dates_to_eras: LiteMap<EraStartDate, TinyStr16>,
+pub struct JapaneseErasV1<'data> {
+    #[cfg_attr(feature = "provider_serde", serde(borrow))]
+    pub dates_to_historical_eras: ZeroVec<'data, (EraStartDate, TinyStr16)>,
```

Comment:
issue: these should both just be ZeroMaps

Reply:
These are not maps conceptually. We need to be able to find the next smaller era start year for any year. From a map we can only get a value for an exact key match.

Reply:
i think they're still logically maps, just ones with a more complex structure. the original code used `get_indexed()` which can be added to `ZeroMap` if necessary. But I guess this works too, there's however no validation of the map being sorted anymore (which might be fine?)

Reply:
@sffc thoughts? over all I'm not strongly in favor of either approach here

Reply:
So far we haven't exposed the internal structure of `ZeroMap`. Adding a `get_indexed` (and `binary_search_by`) will do that, and make them very similar to `ZeroVec<tuple>`.

Reply:
We do actually have some ZeroMap internal structure already exported: the `get_by` function, the `try_append` function, and more generally the requirement that keys are `Ord`. I would support adding `indexed` methods to ZeroMap, separate from the issue at hand.

If ZeroMap had `indexed` methods, would we use it here instead of the tuple vec? I think yes. `LiteMap` exists to improve the ergonomics of `Vec<(K,V)>`. We should do the same here.

Reply:
That being said, I'm also open to the argument the data being expressed here isn't actually a map. It's an interval list, and we should store it like we would store an interval list.